src/core/log/log.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import {
+  type EdgeIo,
+  type EdgeLog,
+  type EdgeLogMethod
+} from '../../types/types.js'
+import { addHiddenProperties } from '../../util/util.js'
+
+type EdgeLogType = 'info' | 'warn' | 'error'
+
+type EdgeLogEntry = {
+  type: EdgeLogType,
+  time: Date,
+  sender: string,
+  message: string
+}
+
+function makeLogMethod(
+  io: EdgeIo,
+  type: EdgeLogType,
+  sender: string
+): EdgeLogMethod {
+  return function log() {
+    let message = ''
+    for (let i = 0; i < arguments.length; ++i) {
+      const arg = arguments[i]
+      if (i > 0) message += ' '
+      message += typeof arg === 'string' ? arg : JSON.stringify(arg)
+    }
+
+    const entry: EdgeLogEntry = { type, time: new Date(), sender, message }
+    io.console.info(entry.sender + ': ' + entry.message)
```

Comment:
I have more commits in this branch, which add an onLog event within the `makeEdgeContext` setup. However, that stuff isn't ready yet. In the mean time, this `EdgeLogEntry` is the format of the event.

Reply:
time?